doc/components.md

```diff
@@ -0,0 +1,221 @@
+# Components
+
+This document provides a full introduction and usage example of each component implemented here. Let's start with it!
+
+## config/aero
+This component is responsible for reading and parsing configuration files from our environment based on a `resources/config.edn` file or getting the current [profile](https://github.com/juxt/aero#profile) on environment var `SYSTEM_ENV`. The used library is defined as `A small library for explicit, intentful configuration` and you can configure multiple environment variables for each specific usage.
+
+Look below for an example of `resources/config.edn` file:
+```clojure
+{:webserver/port #long #or [#env PORT 3001]
+
+ :database {:dbtype "postgres"
+            :dbname #or [#env DB-NAME "postgres"]
+            :username #or [#env DB-USER "postgres"]
+            :password #or [#env DB-PASS "postgres"]
+            :host #or [#env DB-HOST "localhost"]
+            :port #or [#env DB-PORT 5432]}}
+```
+We define a `:webserver/port` for example, that can come from a defined `PORT` in our environment, or if it's not defined, it will assume as `3001`. The same logic is valid for each field in the database configuration.
+
+To start our component we can start building our system map:
+```clojure
+(defn- build-system-map []
+  (component/system-map
+   ;; some other components
+   :config (config/new-config)))
+```
+> It's normal to see this component as the first one on the system map because it will start with every configuration that comes from our environment.
+
+The `dev` profile is the default for `SYSTEM_ENV`, so if you want to set up another profile you can define your `SYSTEM_ENV` to a new value and a new environment will be configured.
+
+## db/jdbc-hikari
+To set up our database that's the component that we use (and it depends on your [connection info data](https://github.com/parenthesin/components/blob/main/test/resources/config.edn#L3) previously configured). This component is using two main libraries to manage our database: [next-jdbc](https://github.com/seancorfield/next-jdbc) and [HikariCP](https://github.com/brettwooldridge/HikariCP).
+
+### next-jdbc
+This library is responsible for managing our access to our database (based on JDBC databases). It is *a new low-level Clojure wrapper for JDBC-based access to databases.* You can check more about [here](https://cljdoc.org/d/com.github.seancorfield/next.jdbc/1.3.955/doc/readme).
```

Comment:
Worth mentioning that we are using [jdcb-url](https://github.com/parenthesin/components/blob/main/src/parenthesin/components/db/jdbc_hikari.clj#L17) to build up the connection string and the db config map, supports all the extra keys as query arguments to the Conn string url like [useSSL](https://github.com/clj-codes/docs.backend/blob/main/resources/config.edn#L12).

[Reference](https://github.com/seancorfield/next-jdbc/blob/develop/doc/getting-started.md#the-db-spec-hash-map)